Sources/Graphiti/Federation/FederationSchema.swift

```diff
@@ -0,0 +1,80 @@
+import Foundation
+import AnyCodable
+
+public protocol FederationEntityKey: Codable {}
+
+public protocol FederationEntity {
+    static var typename: String { get }
+    static var keys: [FederationEntityKey.Type] { get }
+}
+
+public protocol FederationContext {
+    var entities: [FederationEntity.Type] { get }
+    var sdl: String { get }
+}
+
+public struct FederationEntityResolverArguments: Codable {
+    let representations: [AnyCodable]
+}
+
+public protocol FederationResolver {
+    func entity(context: any FederationContext, key: FederationEntityKey) -> FederationEntity?
```

Comment:
Could this be re-written like this?

```swift
public protocol FederationResolver {
    associatedtype Context: FederationContext
    func entity(context: Context, key: FederationEntityKey) -> FederationEntity?
}
```

Reply:
I was getting errors in the `Field` init where the resolver type could not be inferred. But with async resolve it's able to infer it now.